planner/core/rule_build_key_info.go

```diff
@@ -100,6 +101,27 @@ func (p *LogicalSelection) BuildKeyInfo(selfSchema *expression.Schema, childSche
 				p.maxOneRow = true
 				break
 			}
+			// MaxOneRow can be eliminated if there is a unique index in the equivalence condition.
```

Comment:
I prefer to enhance the method `p.checkMaxOneRowCond` to check if the column has a unique index.

Reply:
I agree. I've changed the method `p.checkMaxOneRowCond`.